src/server.rs

```diff
@@ -30,19 +30,13 @@ pub type EhloKeywords = BTreeMap<String, Option<String>>;
 pub type ShutdownSignal = dyn FusedFuture<Output = Result<(), ()>> + Send + Unpin;
 
 #[async_trait]
-pub trait Handler: Send
-where
-    Self::TlsSession: Sync,
-{
+pub trait Handler: Send {
     type TlsConfig;
-    type TlsSession;
 
     async fn tls_request(&mut self) -> Option<Self::TlsConfig> {
         None
     }
 
-    async fn tls_started(&mut self, _session: &Self::TlsSession) {}
```

Comment:
Could you update your patch so that it still allows for inspection of the negotiated TLS parameters ?

Reply:
This was easier than expected.